### ISSUE TITLE
c7n-left - support finer-grained matches against JSON blocks

### ISSUE THREAD

**Issue by ajkerrigan** [cloud-custodian/cloud-custodian#8459]
### Describe the feature

General request: Some configuration blocks in a Terraform resource definition are provided as JSON (either using heredocs or `jsonencode()`). It would be useful to be able to match on their structure rather than relying string matches.

Concrete example: Consider an ECS task definition with container definitions defined as JSON:

```hcl
resource "aws_ecs_task_definition" "plaintext_password" {
  family                = "service"
  container_definitions = <<CONTAINER_DEF
 [
   {
     "name": "my_service",
     "essential": true,
     "memory": 256,
     "environment": [
       { "name": "ENVIRONMENT", "value": "dev" },
       { "name": "API_TOKEN", "value": "ohsosecret123"}
     ]
   }
 ]
 CONTAINER_DEF
}
```

I'd like to be able to look at specific environment variable keys & values, but that requires deserializing the JSON container definitions.

### Extra information or context

- I believe we've had discussions about something like a `value_type: json` transform in the past :thinking: but have typically solved those cases with resource-specific filters.
- For the concrete case today, using string/regex matches against the full JSON blob is often still feasible. It's just a little awkward as we're matching against a string like this with its embedded brackets, braces and newlines:

```console
' [\n   {\n     "name": "my_service",\n     "essential": true,\n     "memory": 256,\n     "environment": [\n       { "name": "ENVIRONMENT", "value": "dev" },\n       { "name": 
"API_TOKEN", "value": "ohsosecret123"}\n     ]\n   }\n ]\n'
```

**Comment by kapilt** [comment]
yeah this also shows up for iam policy documents, I think we want support in list item as well.
